### ISSUE TITLE
Jenkins & dredd. npm ERR! Failed at the protagonist@0.8.0 install script

### ISSUE THREAD

**Issue by imp-partner-test** [apiaryio/dredd#64]
I'm trying to integrate dredd apiary tests into Jenkins. I followed these steps to install NodeJS, npm and dredd on Jenkins (on Linux): 
https://wiki.jenkins-ci.org/display/JENKINS/NodeJS+Plugin

While running the test I got this ERR message: 
npm ERR! Failed at the protagonist@0.8.0 install script.
npm ERR! This is most likely a problem with the protagonist package,
npm ERR! not with npm itself....

Do you know this problem? Is there a solution for it? Thank you.

**Comment by OrCharles** [comment]
I am also experiencing errors with this on OS X 10.10, here is my console output:

```
$ c++ --version
Apple LLVM version 6.0 (clang-600.0.56) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin14.1.0
Thread model: posix

$ echo $CC

```
nothing is outputted for $CC but I do get the same output w/r/t protagonist/node-gyp as shown above:
```
npm ERR! Darwin 14.1.0
npm ERR! argv "node" "/usr/local/bin/npm" "update"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1
npm ERR! code ELIFECYCLE

npm ERR! protagonist@0.17.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the protagonist@0.17.1 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the protagonist package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls protagonist
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/cwisniewski/Documents/dev/git/api/tests/node/npm-debug.log
```
Should I reach out here as well? https://github.com/apiaryio/protagonist/issues

**Comment by netmilk** [comment]
@OrCharles thanks a lot for the report and I'm sorry about any inconvenience. I think this is more likely a Protagonist related issue, so report it there, please.

Protagonist is a wrapper around Snowcrash API Blueprint parser, so you can try to install Snowcrash directly on osx:

```bash
$ brew install --HEAD \
  https://raw.github.com/apiaryio/snowcrash/master/tools/homebrew/snowcrash.rb
```

If this will fail, it's definitely a problem with installation of Protagonist.

Btw: my osx system outputs this:
```bash
$ echo $CC
gcc-4.2
```
Maybe figure this out and it will work.

**Comment by bugthing** [comment]
i can not `npm install dredd` as it depends on protagonist 0.17 and node 0.12.2 (on arch linux) does not build this .. it will however build protagonist 0.19 .. therefore it would be super rad if dredd depended on version 0.19 of protagonist

not sure why, etc., just reporting my findings

**Comment by kylef** [comment]
This issue is due to installing dredd with NPM on NodeJS v0.12. I experienced it before, the current release of dredd requires NodeJS v0.10 due to depending on an older protagonist that depends on NodeJS v0.10.

**TL;DR**: *dredd 0.5.0 will require NodeJS v0.10 instead of v0.12 (latest).*

**Comment by netmilk** [comment]
Dredd supports node v0.10, node v0.12 and io.js as well for some time already. Closing this one, please reopen or submit new issue if you'll experience any further inconvenience regarding this.

**Comment by HansHauge** [comment]
For what it's worth we just had this issue. We ended up needing to update gcc on our Jenkins server.

**Comment by honzajavorek** [comment]
@HansHauge I believe it's a different issue. Recent changes cause that to be able to install newer Dredd, one will need a C++ compiler able to compile C++11. In the future we will eventually migrate to js-only parser by default and no compiler will be needed, but as of now, this change is needed in order to make Dredd work with your CI (especially Travis CI config files need changes - see e.g. https://github.com/apiaryio/dredd-hooks-python/pull/19).

Details on why all this: https://github.com/apiaryio/protagonist/blob/master/CHANGELOG.md#breaking
